### DISCUSSION TITLE
Can't use mvvm to bind

### DISCUSSION THREAD

**Discussion by MarkoShenodaa** [AvaloniaUI/Avalonia#15773] [Q&A]
### Describe the bug

Unable to resolve property or method of name 'ClickCommand' on type 'AvaloniaApplication2.ViewModels.MainViewModel'.

I start new Avalonia App as it is just add one more property and one command to test.

here is the code 

<UserControl
    x:Class="AvaloniaApplication2.Views.MainView"
    xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:vm="using:AvaloniaApplication2.ViewModels"
    d:DesignHeight="450"
    d:DesignWidth="800"
    x:DataType="vm:MainViewModel"
    mc:Ignorable="d">
    <UserControl.DataContext>
        <vm:MainViewModel />
    </UserControl.DataContext>
    <StackPanel HorizontalAlignment="Center" VerticalAlignment="Center">
        <TextBlock
            HorizontalAlignment="Center"
            VerticalAlignment="Center"
            Text="{Binding Greeting}" />
        <Button
            Margin="10"
            HorizontalAlignment="Center"
            VerticalAlignment="Center"
            Command="{Binding ClickCommand}"
            Content="{Binding ButtonTitle}" />
    </StackPanel>
</UserControl>

public partial class MainViewModel : ViewModelBase
{
    public MainViewModel()
    {
        ButtonTitle = "Click me";
    }
    private int count = 0;

    [ObservableProperty]
    private string _greeting = "Welcome to Avalonia!";

    [ObservableProperty]
    private string _buttonTitle ;

    [RelayCommand]
    public void Click()
    {
        count++;

        if (count == 1) 
        {
            ButtonTitle = $"You Clicked Me {count} Time";
            return; 
        }

        ButtonTitle = $"You Clicked Me {count} Times";
    }
}

### To Reproduce

just start new project and add the property and the command 

### Avalonia version

11.0.2

**Comment by stevemonaco** [comment]
This almost certainly isn't a bug and should be moved to discussions.

But what XAML are you using? Particularly, the entirety of your `<Button ... />`

**Comment by thevortexcloud** [comment]
I am guessing this is a casting problem. In which case you can either cast to your view model type or use reflection binding.

https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings#type-casting
